packages/react-flight-dom-relay/src/__tests__/ReactFlightDOMRelay-test.internal.js

```diff
@@ -53,6 +41,81 @@ describe('ReactFlightDOMRelay', () => {
     }
     let model = ReactDOMFlightRelayClient.getModelRoot(response).model;
     ReactDOMFlightRelayClient.close(response);
-    expect(model).toEqual({foo: {bar: ['A', 'B']}});
+    return model;
+  }
+
+  function block(query, render) {
+    return function(...args) {
+      let curriedQuery = () => {
+        return query(...args);
+      };
+      return [Symbol.for('react.server.block'), render, curriedQuery];
+    };
+  }
+
+  it('can render a server component', () => {
+    function Bar({text}) {
+      return text.toUpperCase();
```

Comment:
Is returning arrays from these supposed to work? I tried wrapping it in `[]` and got `Cannot assign to read only property 'children' of object '#<Object>'`

Reply:
Hmm actually this only happens when my `toEqual` fails. Maybe something's wrong with some test helper?

Reply:
This is a Jest bug introduced in 0.25.0 https://github.com/facebook/jest/issues/9531

Reply:
Fixed in master but not yet released https://github.com/facebook/jest/pull/9575

Reply:
There's more work to do on server components. If anything it's too permissive and allows anything to serialize here. It should take a separate path that only allows certain renderable things, same as the client.

Somewhat surprising that it doesn't allow an array though.

Reply:
Actually I think that particular error actually just a problem with jest logging/matching. When things are not toEqual which includes elements, that are frozen, for some reason there's a write somewhere. So if it doesn't match you get a completely confusing error.

You probably just didn't update the toEqual case to match.

Reply:
Yeah you're right.